engine.py

```diff
@@ -66,50 +61,163 @@ def create_shotgun_menu(self):
     def has_ui(self):
+        """
+        Indicates that the host application that the engine is connected to has a UI enabled.
+        This always returns False for some engines (such as the shell engine) and may vary
+        for some engines, depending if the host application for example is in batch mode or
+        UI mode.
+        :returns: boolean value indicating if a UI currently exists
+        """
         return True
 
     ##########################################################################################
     # logging
 
     def _emit_log_message(self, handler, record):
-        if record.levelno < logging.INFO:
-            formatter = logging.Formatter("Debug: Shotgun %(basename)s: %(message)s")
-        else:
-            formatter = logging.Formatter("Shotgun %(basename)s: %(message)s")
-
-        msg = formatter.format(record)
-
-        print msg
+        """
+        Called by the engine whenever a new log message is available.
+        All log messages from the toolkit logging namespace will be passed to this method.
+        .. note:: To implement logging in your engine implementation, subclass
+                  this method and display the record in a suitable way - typically
+                  this means sending it to a built-in DCC console. In addition to this,
+                  ensure that your engine implementation *does not* subclass
+                  the (old) :meth:`Engine.log_debug`, :meth:`Engine.log_info` family
+                  of logging methods.
+                  For a consistent output, use the formatter that is associated with
+                  the log handler that is passed in. A basic implementation of
+                  this method could look like this::
+                      # call out to handler to format message in a standard way
+                      msg_str = handler.format(record)
+                      # display message
+                      print msg_str
+        .. warning:: This method may be executing called from worker threads. In DCC
+                     environments, where it is important that the console/logging output
+                     always happens in the main thread, it is recommended that you
+                     use the :meth:`async_execute_in_main_thread` to ensure that your
+                     logging code is writing to the DCC console in the main thread.
+        :param handler: Log handler that this message was dispatched from
+        :type handler: :class:`~python.logging.LogHandler`
+        :param record: Std python logging record
+        :type record: :class:`~python.logging.LogRecord`
+        """
+        log_debug = record.levelno < logging.INFO and sgtk.LogManager().global_debug
+        log_info_above = record.levelno >= logging.INFO
+        if log_debug or log_info_above:
+            msg = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S,%f")[:-3]
+            msg += " "
+            msg += handler.format(record)
+            print msg
```

Comment:
Might want to add
```python
from __future__ import print_function
```

up top, then

```suggestion
            print(msg, handler.format(record))
```